libr/core/cmd_api.c

```diff
@@ -53,10 +90,39 @@ R_API RCmd *r_cmd_free(RCmd *cmd) {
 			R_FREE (cmd->cmds[i]);
 		}
 	}
+	free_cmd_desc_tree (cmd->root_cmd_desc);
 	free (cmd);
 	return NULL;
 }
 
+R_API RCmdDesc *r_cmd_get_root(RCmd *cmd) {
+	return cmd->root_cmd_desc;
+}
+
+static bool is_valid_desc(RCmdDesc *cd, const char *name) {
+	return cd->type == R_CMD_DESC_TYPE_OLDINPUT ||
+		(cd->type == R_CMD_DESC_TYPE_ARGV && !strcmp (cd->name, name));
```

Comment:
Why is strcmp required? I can imagine situations where it would be useful to share the callback and differentiate the modes using command name.

Reply:
I think for that we may need a different type in the future, otherwise how do you differentiate when the CmdDesc should handle it and when it should not?

e.g.

`afllll` should not be handled by `afl`, it should just say "command not found" (so r_cmd_get_desc should return NULL).

Maybe I can make this a little bit more clear, let me try to refactor this piece of code.

Reply:
Sounds reasonable. I agree that having separate type or a flag for that would be better if it helps detecting wrong command.